rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1179,175 +1170,191 @@ public NodeState getNodeState(String nodeUrl) {
+                                                              nodeSource.getStatus().toString()));
+
+        logger.info("Node source " + nodeSourceName + " has been successfully defined");
+
+        return new BooleanWrapper(true);
+    }
+
+    private NodeSourceDescriptor persistNodeSourceAndGetDescriptor(String nodeSourceName, String infrastructureType,
+            Object[] infraParams, String policyType, Object[] policyParams, boolean nodesRecoverable) {
+
+        List<Serializable> serializableInfraParams = this.getSerializableParamsOrFail(nodeSourceName, infraParams);
+        List<Serializable> serializablePolicyParams = this.getSerializableParamsOrFail(nodeSourceName, policyParams);
+
         NodeSourceData persistedNodeSource = new NodeSourceData(nodeSourceName,
                                                                 infrastructureType,
                                                                 serializableInfraParams,
                                                                 policyType,
                                                                 serializablePolicyParams,
-                                                                caller,
+                                                                this.caller,
                                                                 nodesRecoverable,
                                                                 NodeSourceStatus.NODES_UNDEPLOYED);
 
-        boolean added = dbManager.addNodeSource(persistedNodeSource);
+        boolean success = this.dbManager.addNodeSource(persistedNodeSource);
 
-        if (!added) {
-            return new BooleanWrapper(false);
-        } else {
-            return validateAndPrepareNodeSource(nodeSourceName, persistedNodeSource, added);
+        if (!success) {
+            this.dbManager.removeNodeSource(nodeSourceName);
         }
 
+        return persistedNodeSource.toNodeSourceDescriptor();
     }
 
-    private List<Serializable> getSerializableParamsOrFail(String nodeSourceName, Object[] params) {
-        List<Serializable> serializableParams = null;
-        if (params != null) {
-            Object currentParam = null;
-            try {
-                serializableParams = new ArrayList<>(params.length);
-                for (int i = 0; i < params.length; i++) {
-                    currentParam = params[i];
-                    serializableParams.add(i, (Serializable) currentParam);
-                }
-            } catch (ClassCastException e) {
-                throw new IllegalArgumentException("Parameter " + currentParam + " of defined node source " +
-                                                   nodeSourceName + " is not Serializable", e);
-            }
-        }
-        return serializableParams;
-    }
+    private List<Serializable> getSerializableParamsOrFail(String nodeSourceName, Object[] parameters) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "nodeSourceName". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

Reply:
Done.